.github/workflows/ci.yml

```diff
@@ -1,43 +1,34 @@
-name: Test
+name: Ruby
 
 on:
   push:
     branches: [main, master] # TODO: rename and get rid of 'master'
-  # Triggers the workflow on pull request events.
+    paths-ignore:
+      - '**/*.md'
   pull_request:
-    types: [opened, reopened, synchronize]
-  # Allows you to run this workflow manually from the Actions tab
+    branches: [main, master] # TODO: rename and get rid of 'master'
+    paths-ignore:
+      - '**/*.md'
   workflow_dispatch:
 
-# This allows a subsequently queued workflow run to interrupt previous runs.
-concurrency:
-  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
-  cancel-in-progress: true
-
 jobs:
   test:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby:
-          # https://www.ruby-lang.org/en/downloads/branches/
-          - "ruby-3.0"
-          - "ruby-3.1"
-          - "ruby-3.2"
-          - "ruby-3.3"
-          - "ruby-head"
-          # https://github.com/jruby/jruby/discussions/7915
-          - "jruby-9.4"
-          - "jruby-head"
-    name: "Test (Ruby ${{ matrix.ruby }})"
-    env:
-      SKIP_ADAPTERS: jr_jackson,nsjsonserialization,json_pure
+        ruby-version: ['3.0', '3.1', '3.2', 'jruby']
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+
```

Comment:
I refactored this, easier to read. I also added macos and windows to the platform list, and got rid of the `SKIP_ADAPTERS` env since you're not using it any more.